### PR TITLE
RHMAP-2301 -  public ssh keys added to authorized keys file.

### PR DESCRIPTION
**Pull request by wtrocki**
###  What was done
- Added script to execute actions before starting sshd
- Minor changes in docker image.

## Requirements 
Image require secrets to be mounted locally. 
See openshift templates PR for more details.

### Testing 
ssh -i gitlab_shell_prv git@$(oc get service gitlab-shell --no-headers | awk '{print $2}')

See jira for verification steps.
Review: @david-martin @maleck13

### CODE REVIEW

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.
Fhcap PR will be generated automatically.
To deploy your changes to a cluster place comment in this PR with pattern [x][y] where x=deploy and y=clustername you wish to deploy. 
You can also deploy your change manually using fhcap-cli

File: bin/start.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+SSH_SECRETS_DIR=${SSH_SECRETS_DIR:-/etc/feedhenry/gitlab-shell}
+AUTHORIZED_KEYS_FILE=${AUTHORIZED_KEYS_FILE:-/home/git/.ssh/authorized_keys} 
+
+echo "Initializing authorized_keys file"
+cat $SSH_SECRETS_DIR/repoadmin-id-rsa-pub > $AUTHORIZED_KEYS_FILE
```
Comment (wtrocki) [inline bin/start.sh]: @david-martin I'm intentionally cleaning all authorized keys here to make sure that initially only platform has access to that daemon.

Comment (david-martin) [inline bin/start.sh]: Does this mean the file will be reset any time the container restarts or is rescheduled?

Comment (david-martin) [inline bin/start.sh]: For cheffed clusters, we only add the keys if they aren't already there https://github.com/fheng/fhcap/blob/master/wrapper-cookbooks/gitlabshell_feedhenry/recipes/default.rb#L35-L63

**Comment by wtrocki** [comment]
Retest this please

File: Dockerfile [hunk 2]
```diff
@@ -11,21 +11,28 @@ RUN ["bash", "-c", "yum install -y --setopt=tsflags=nodocs openssh-server libicu
...
-                     ./bin/install && \
-                     chown -R git:git /home/git"]
+RUN mkdir /home/git/gitlab-config && \
+    ## Setup default config placeholder
+    cp config.yml.example ../gitlab-config/config.yml && \
+    ln -s /home/git/gitlab-config/config.yml && \
+    # PAM workarounds for docker and public key auth
+    sed -i \
+          -e "s|session\s*required\s*pam_loginuid.so|session optional pam_loginuid.so|g" \
+          -e "s|account\s*required\s*pam_nologin.so|#account optional pam_nologin.so|g" \
```
Comment (maleck13) [inline Dockerfile]: So this stuff doesn't mean a lot to me. I think a link to some doc explanation would be good so we understand what is happening here

Comment (wtrocki) [inline Dockerfile]: Added some useful reference. RHEL by default reject non root users from using sshd 🐼 
http://man7.org/linux/man-pages/man8/pam_nologin.8.html

**Comment by wtrocki** [comment]
Additional changes: Set fixed key names. 

PR verified in as part of gitlab template testing (fh-openshift-templates). 
Version published to dockerhub: 2.0.0.pre-29

**Pull request merged**